### DISCUSSION TITLE
DataLoader support for Webflux

### DISCUSSION THREAD

**Discussion by mlsoftwarehouse** [Netflix/dgs-framework#774] [Q&A]
Hi!
I am tying to implement my GraphQL endpoint using Webflux.
**I am struggling to use DataLoader** as described under: https://netflix.github.io/dgs/data-loaders/

My DataFetcher is declared as follows:
`@DgsData(parentType = DgsConstants.NAVIGATIONITEM.TYPE_NAME, field = DgsConstants.NAVIGATIONITEM.Children)
  public Flux<NavigationItem> children(DataFetchingEnvironment environment) {`

However the DataLoaders#load method return type is 
`public CompletionStage<List<NavigationItem>> load(List<Long> keys, BatchLoaderEnvironment batchLoaderEnvironment) {`

How i can make both to work together?
Could you please point me to the right direction?

**Comment by paulbakker** [comment]
You can create a Mono/Flux from a `CompletableFuture`. For example: https://projectreactor.io/docs/core/release/api/reactor/core/publisher/Mono.html#fromFuture-java.util.concurrent.CompletableFuture-.

**Reply by mlsoftwarehouse** [reply]
I've tried different approaches to convert Flux to Future but it didn't work. It compiles but every time the result of data loader is null (even thought I can see the DB is being hit and the result of the loader is there). Do you recall any working example of DataLoader + Webflux?
